dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/BaseClasses.mo

```diff
@@ -54,13 +53,196 @@ package BaseClasses
+    Types.ReactivePowerPu QInj1RawPu "Reactive power generation of converter 1 without taking limits into account in pu (base SnRef) (generator convention)";
+
+    annotation(preferredView = "text");
+  end BasePQPropDangling;
+
+  partial model BasePQProp "Base dynamic model for proportional reactive power control"
+    extends BaseClasses.BasePQPropDangling;
+
+    parameter Real QPercent2 "Percentage of the coordinated reactive control that comes from converter 2";
+
+    input Types.PerUnit NQ2 "Signal to change the reactive power of converter 2 depending on the centralized voltage regulation (generator convention)";
+
+    Types.ReactivePowerPu QInj2RawModeUPu "Reactive power generation of converter 2 without taking limits into account in pu and for mode U activated (base SnRef) (generator convention)";
+    Types.ReactivePowerPu QInj2RawPu "Reactive power generation of converter 2 without taking limits into account in pu (base SnRef) (generator convention)";
+
+    annotation(preferredView = "text");
+  end BasePQProp;
+
+  partial model BasePTanPhiDangling "Base dynamic model for P/tan(Phi) control at terminal 1"
+
+    input Real tanPhi1Ref(start = TanPhi1Ref0) "tan(Phi) regulation set point at terminal 1";
+
+    parameter Real TanPhi1Ref0 "Start value of tan(Phi) regulation set point at terminal 1";
+
+  protected
+    Types.ReactivePowerPu QInj1RawPu "Raw reactive power at terminal 1 in pu (base SnRef) (generator convention)";
+
+    annotation(preferredView = "text");
+  end BasePTanPhiDangling;
+
+  partial model BasePTanPhi "Base dynamic model for P/tan(Phi) control"
+    extends BaseClasses.BasePTanPhiDangling;
+
+    input Real tanPhi2Ref(start = TanPhi2Ref0) "tan(Phi) regulation set point at terminal 2";
+
+    parameter Real TanPhi2Ref0 "Start value of tan(Phi) regulation set point at terminal 2";
+
+  protected
+    Types.ReactivePowerPu QInj2RawPu "Raw reactive power at terminal 2 in pu (base SnRef) (generator convention)";
+
+    annotation(preferredView = "text");
+  end BasePTanPhi;
+
+  partial model BasePVDangling "Base dynamic model for PV control at terminal 1"
+
+    parameter Boolean UseLambda1 = not(Lambda1Pu == 0) "If true, the voltage regulation follows the law U1RefPu = U1Pu + Lambda1Pu * QInj1Pu at terminal 1";
+    parameter Types.PerUnit Lambda1Pu "Parameter Lambda of the voltage regulation law U1RefPu = U1Pu + Lambda1Pu * QInj1Pu, in pu (base UNom, SnRef) at terminal 1";
+    parameter Types.ReactivePower Q1Nom "Nominal reactive power in Mvar at terminal 1";
+
+    input Types.VoltageModulePu U1RefPu(start = U1Ref0Pu) "Voltage regulation set point in pu (base UNom) at terminal 1";
+
+    parameter Types.VoltageModulePu U1Ref0Pu "Start value of the voltage regulation set point in pu (base UNom) at terminal 1";
+
+    Types.ReactivePowerPu QInj1PuQNom "Reactive power at terminal 1 in pu (base Q1Nom) (generator convention)";
```

Comment:
This is no initial variable, so this line has to be moved between U1RefPu and U1Ref0Pu.